### PR TITLE
[MI-2159] Test cases for some components

### PR DESCRIPTION
**Pull request by ayusht2810**
Created test cases for components:

- download_button
- recordTypePanel
- subscriptionTypePanel
- eventsPanel

### CODE REVIEW

File: webapp/package.json [hunk 1]
```diff
@@ -66,14 +66,16 @@
...
     "@reduxjs/toolkit": "1.8.3",
-    "bootstrap": "5.1.3",
+    "@types/redux-mock-store": "1.0.3",
+    "bootstrap": "4.6.2",
     "core-js": "3.6.5",
     "js-cookie": "3.0.1",
     "mattermost-redux": "5.27.0",
     "react": "16.13.1",
-    "react-bootstrap": "2.4.0",
+    "react-bootstrap": "1.0.0",
```
Comment (sooraj-shukla) [inline webapp/package.json:75]: Why we degraded the packages?

File: webapp/src/components/admin_settings/download_button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+describe('Download Button', () => {
+    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
+
+    beforeEach(() => {
+        component = shallow(<DownloadButtons {...downloadButtonProps}/>);
+    });
+
+    afterEach(() => {
+        component.unmount();
```
Comment (sooraj-shukla) [inline webapp/src/components/admin_settings/download_button.test.tsx:28]: Do we really need to call unmount after each test if we're doing shallow rendering?

File: webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        const store = mockStore(initialState);
+        component = shallow(
+            <Provider store={store}>
+                <AddSubscriptions/>
+            </Provider>,
+        );
+    });
+
+    afterEach(() => {
+        component.unmount();
```
Comment (sooraj-shukla) [inline webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx:24]: Same as above.

File: webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    });
+
+    it('Should render correctly', () => {
+        expect(component).toMatchSnapshot();
+    });
+
+    it('Should render the edit subscription body correctly', () => {
```
Comment (sooraj-shukla) [inline webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx:31]: Aren't we testing for add subscription here? I guess it should be `Should render the add subscription body correctly`

File: webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        expect(component.find('AddOrEditSubscriptionModal')).toBeTruthy();
+    });
+});
```
Comment (sooraj-shukla) [inline webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx:34]: I guess we should also add a test to check whether the `editSubscriptionData`(`pluginState.openEditSubscriptionModalReducer.data` the data passed as `subscriptionData` prop) is falsy, if possible.

File: webapp/src/containers/addOrEditSubscriptions/editSubscription/editSubscription.test.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+    afterEach(() => {
+        component.unmount();
+    });
+
+    it('Should render correctly', () => {
+        expect(component).toMatchSnapshot();
+    });
+
+    it('Should render the edit subscription body correctly', () => {
```
Comment (sooraj-shukla) [inline webapp/src/containers/addOrEditSubscriptions/editSubscription/editSubscription.test.tsx:31]: I guess we should also add a test to check whether the `subscriptionData` is correctly passed and is not null, if possible.

File: webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.test.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        expect(component).toMatchSnapshot();
+    });
+
+    it('Should apply the passed className prop', () => {
+        expect(component.hasClass(eventsPanelProps.className)).toBeTruthy();
+    });
+
+    it('Should render the events panel body correctly', () => {
+        expect(component.find('Checkbox')).toHaveLength(5);
+        expect(component.find('ModalSubtitleAndError')).toHaveLength(1);
```
Comment (sooraj-shukla) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.test.tsx:60]: Shouldn't we also test that `ModalSubstitleAndError` isn't rendered if the error is not passed.

File: webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.test.tsx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        )).toBeTruthy();
+    });
+
+    it('Should render the footer correctly', () => {
+        expect(component.contains(
+            <ModalFooter
+                onHide={eventsPanelProps.onBack}
+                onConfirm={eventsPanelProps.onContinue}
+                confirmBtnText='Continue'
+                cancelBtnText='Back'
```
Comment (sooraj-shukla) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.test.tsx:110]: I guess we should also test the footer functionality as well, like check when the buttons are disabled the handlers shouldn't be called when these are enabled the handlers should be called, at least for one of the components.

**Review by sooraj-shukla — CHANGES_REQUESTED** [review]

File: webapp/package.json (hunk 1 above)
Comment (ayusht2810) [reply webapp/package.json:75]: we were getting some errors like `react jsx/runtime` and some type errors. we needed to upgrade either react version or downgrade these packages. so downgraded them

File: webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx (hunk 4 above)
Comment (ayusht2810) [reply webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx]: removed these test files as on mounting, we were getting states as undefined

File: webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx (hunk 5 above)
Comment (ayusht2810) [reply webapp/src/containers/addOrEditSubscriptions/addSubscription/addSubscription.test.tsx]: same

File: webapp/src/containers/addOrEditSubscriptions/editSubscription/editSubscription.test.tsx (hunk 6 above)
Comment (ayusht2810) [reply webapp/src/containers/addOrEditSubscriptions/editSubscription/editSubscription.test.tsx]: same

File: webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.test.tsx (hunk 8 above)
Comment (ayusht2810) [reply webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.test.tsx:106]: i don't think we need to test these functionality as it is already been tested in the mm UI library

File: webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.test.tsx (hunk 7 above)
Comment (ayusht2810) [reply webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.test.tsx:56]: tried it, but this component will always be present, it is just rendering on basis of error

Comment (ayusht2810) [reply webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.test.tsx:56]: updated this

**Review by sooraj-shukla — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Comment by manojmalik20** [comment]
Also check, if any snapshot needs to be updated.
